dvc/scm/__init__.py

```diff
@@ -21,3 +23,20 @@ def SCM(root_dir, repo=None):  # pylint: disable=invalid-name
         return Git(root_dir, repo=repo)
 
     return Base(root_dir, repo=repo)
+
+
+def scm_context(method):
+    def run(*args, **kw):
+        repo = args[0]
+        assert type(repo) is dvc.repo.Repo
+        scm = repo.scm
+        try:
+            result = method(*args, **kw)
+            scm.reset_ignores()
+            repo.remind_to_git_add()
```

Comment:
Let's move `remind_to_git_add()` along with `repo.files_to_git_add` into scm instead of keeping it in the repo. That should fit nicely 🙂